### ISSUE TITLE
ISO 10303-52: Index needs to support different types of content

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-iso#605]
In ISO 10303-52, there are different types of index terms encoded in the Index:

<img width="495" alt="Screenshot 2022-04-02 at 2 48 23 PM" src="https://user-images.githubusercontent.com/11865/161370850-eac7630f-dccb-46df-8023-ca9f3f916d49.png">
<img width="486" alt="Screenshot 2022-04-02 at 2 48 33 PM" src="https://user-images.githubusercontent.com/11865/161370855-8991e3ef-7e71-4af3-a870-f5005dd5773f.png">

* Index terms on their own are general English terms
* Index terms with a suffix of "(entity)" means they are EXPRESS entities
* Index terms with a suffix of "(function)" means they are EXPRESS functions
* Index terms with a suffix of "(type)" means they are EXPRESS data types

We need a way to specify these in the markup.

**Comment by opoudjis** [comment]
> We need a way to specify these in the markup.

No, we don't. We need a way of inserting these suffixes, "(entity)", "(function)", "(type)" into the index term markup, depending on the type of thing being indexed.

**Comment by opoudjis** [comment]
All that is needed here is the equivalent of the following, in markup generated by Liquid:

https://github.com/metanorma/annotated-express/blob/main/data/documents/resources/fundamentals_of_product_description_and_support/sections/04-schemas.adoc

`==== {{ function.id }} ((({{ function.id }},Object definition)))` => `==== {{ function.id }} ((({{ function.id }} (function),Object definition)))`

`==== {{ entity.id }} <<expg.{{ schema.id }},image:expg.gif[]>> ((({{ entity.id }},Object definition)))` => `==== {{ entity.id }} <<expg.{{ schema.id }},image:expg.gif[]>> ((({{ entity.id }} (entity),Object definition)))`

`==== {{ type.id }} <<expg.{{ schema.id }},image:expg.gif[]>> ((({{ type.id }},Object definition)))` => `==== {{ type.id }} <<expg.{{ schema.id }},image:expg.gif[]>> ((({{ type.id }} (type),Object definition)))`

Why on earth wouldn't this be enough?

**Comment by ronaldtse** [comment]
Agree, for the current moment. Thanks.
